mail_sendgrid/tests/test_mail_sendgrid.py

```diff
@@ -0,0 +1,233 @@
+# -*- coding: utf-8 -*-
+# © 2017 Emanuel Cino - <ecino@compassion.ch>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import json
+
+import mock
+from odoo.tests.common import HttpCase
+from ..controllers.json_request import RESTJsonRequest
+
+mock_base_send = 'odoo.addons.mail.models.mail_mail.MailMail.send'
+mock_sendgrid_api_client = ('odoo.addons.mail_sendgrid.models.mail_mail'
+                            '.SendGridAPIClient')
+mock_sendgrid_send = ('odoo.addons.mail_sendgrid.models.mail_mail.'
+                      'OdooMail.send_sendgrid')
+mock_config = ('odoo.addons.mail_sendgrid.models.mail_mail.'
+               'config')
+
+mock_json_request = 'odoo.http.Root.get_request'
+
+
+def side_effect_json(http_request):
+    return RESTJsonRequest(http_request)
+
+
+class FakeClient(object):
+    """ Mock Sendgrid APIClient """
+    status_code = 202
+    body = 'ok'
+
+    def __init__(self):
+        self.client = self
+        self.mail = self
+        self.send = self
+
+    def post(self, **kwargs):
+        return self
+
+
+class FakeRequest(object):
+    """ Simulate a Sendgrid JSON request """
+    def __init__(self, data):
+        self.jsonrequest = [data]
+
+
+class TestMailSendgrid(HttpCase):
+    def setUp(self):
```

Comment:
if possible use `setUpClass` to create records that are always used: boosts test perf a lot ;)

Reply:
Not working with HttpCase? I get errors that cls.env is not defined